metrics/report/report_dockerfile/mem-in-cont.R

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env Rscript
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Analyse the runtime component memory footprint data.
+
+library(ggplot2)					# ability to plot nicely.
+							# So we can plot multiple graphs
+library(gridExtra)					# together.
+suppressMessages(suppressWarnings(library(ggpubr)))	# for ggtexttable.
+suppressMessages(library(jsonlite))			# to load the data.
+
+testnames=c(
+	"memory-footprint-inside-container"
+)
+
+data=c()
+rstats=c()
+rstats_rows=c()
+rstats_cols=c()
+
+# For each set of results
+for (currentdir in resultdirs) {
+	dirstats=c()
+	# For the two different types of memory footprint measures
+	for (testname in testnames) {
+		# R seems not to like double path slashes '//' ?
+		fname=paste(inputdir, currentdir, testname, '.json', sep="")
+		if ( !file.exists(fname)) {
+			warning(paste("Skipping non-existent file: ", fname))
+			next
+		}
+
+		# Derive the name from the test result dirname
+		datasetname=basename(currentdir)
+
+		# Import the data
+		fdata=fromJSON(fname)
+		fdata=fdata[[testname]]
+		# Copy the average result into a shorter, more accesible name
+		fdata$requested=fdata$Results$memrequest$Result
+		fdata$total=fdata$Results$memtotal$Result
+		fdata$free=fdata$Results$memfree$Result
+		fdata$avail=fdata$Results$memavailable$Result
```

Comment:
`this$still$makes$my$eyes$bleed$every$time$I$see$it $)`

Reply:
:-)
 I could whitespace it more at least, like:
```R
fdata$avail = fdata$Results$memavailable$Result
```

But, the `$`s are R's way of accessing the dataframe members I'm afraid, and the long strong on the RHS is what comes in from the JSON import, so I don't have a huge amount of control over that.

Yeah, they reckon the 80% (of maybe the 80/20 rule) of data analysis is crunching the data into the right format to then be able to process/show it...